### PR TITLE
Upload files

### PR DESCRIPTION
**Pull request by AELSchauer**
This adds paperclip to our app, as well as the upload functionality. However, there are no tests for it yet. We need to look into that more and add them.

@cmacaulay @JonathanSR @alfosco

### CODE REVIEW

File: app/controllers/uploads_controller.rb [hunk 1]
```diff
@@ -1,5 +1,27 @@
 class UploadsController < ApplicationController
+
```
Comment (AELSchauer) [inline app/controllers/uploads_controller.rb]: Done

File: app/controllers/uploads_controller.rb [hunk 2]
```diff
@@ -1,5 +1,27 @@
...
+  def new
+    @upload = Upload.new
+  end
+
+  def create
+    @upload = Upload.new(upload_params)
+    if @upload.save
+      flash[:success] = "Your file has been uploaded!"
+      redirect_to home_path
```
Comment (AELSchauer) [inline app/controllers/uploads_controller.rb]: Ignore this.

File: app/controllers/uploads_controller.rb [hunk 3]
```diff
@@ -1,5 +1,27 @@
...
+    else
+      flash[:danger] = "Please try uploading again"
+      redirect_to home_path
```
Comment (AELSchauer) [inline app/controllers/uploads_controller.rb]: Ignore this.

File: app/models/upload.rb [hunk 4]
```diff
@@ -1,11 +1,15 @@
...
+  alias_attribute :content_type, :file_content_type
+  alias_attribute :size, :file_file_size
 
-  def file_type
-    "PDF"
+  def user
+    folder.user
   end
 end
+p
```
Comment (AELSchauer) [inline app/models/upload.rb]: Done.

File: db/schema.rb [hunk 5]
```diff
@@ -49,11 +49,14 @@
   end
 
   create_table "uploads", force: :cascade do |t|
-    t.integer  "status",     default: 0
-    t.string   "name"
+    t.integer  "status",            default: 0
```
Comment (AELSchauer) [inline db/schema.rb]: Need to fix so that db/schema.rb is ignored.

File: db/schema.rb [hunk 6]
```diff
@@ -10,7 +10,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 20170329002150) do
+ActiveRecord::Schema.define(version: 20170329195624) do
```
Comment (AELSchauer) [inline db/schema.rb]: Need to fix so that db/schema.rb is ignored.

File: app/models/folder.rb [hunk 7]
```diff
@@ -21,4 +21,12 @@ def children
...
     children.sort_by(&:name)
   end
+
+  def content_type
+    ""
+  end
+
+  def size
+    ""
+  end
```
Comment (cmacaulay) [inline app/models/folder.rb]: Still not loving this.

Comment (JonathanSR) [inline app/models/folder.rb]: I dont understand what is going here @cmacaulay, can you explain?

**Pull request merged**